python-package/lightgbm/sklearn.py

```diff
@@ -656,6 +656,7 @@ def fit(self, X, y,
             eval_class_weight=None, eval_init_score=None, eval_metric=None,
             early_stopping_rounds=None, verbose=True,
             feature_name='auto', categorical_feature='auto', callbacks=None):
+        _LGBMAssertAllFinite(y)
```

Comment:
Is this check for LightGBM classification `y` OK?

https://github.com/scikit-learn/scikit-learn/blob/a11154e727ca17be64090083849c564eb4ffbeac/sklearn/utils/validation.py#L59

>Throw a ValueError if X contains NaN or infinity.

P.S. we check `y`, not `X`.